### PR TITLE
Swipe to refresh

### PR DESCRIPTION
**Pull request by shaus019**

### CODE REVIEW

File: app/src/main/java/com/sthoray/allright/ui/main/viewmodel/MainViewModel.kt [hunk 1]
```diff
@@ -35,6 +35,10 @@ class MainViewModel(
         getFeaturedCategories()
         getSecondTierCategories()
     }
+    /** public functions, so we can access it from somewhere else. */
+    fun getRefreshFragment(){
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/ui/main/viewmodel/MainViewModel.kt:39]: If I didn't know the purpose of this function in advanced I would need to see what it does. Please try to think of a more descriptive name along with a doc comment to describe the purpose.

File: app/src/main/java/com/sthoray/allright/ui/search/view/ResultsFragment.kt [hunk 2]
```diff
@@ -42,6 +42,14 @@ class ResultsFragment : Fragment(R.layout.fragment_results) {
         setupFab()
         setListingOnClickListeners()
         setupObservers()
+        swipeToRefreshResultFragment()
+    }
+
+    private fun swipeToRefreshResultFragment() {
+        swipeRefreshResult.setOnRefreshListener {
+            viewModel.applyFiltersAndSearch()
+            swipeRefreshResult.isRefreshing = false
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/ui/search/view/ResultsFragment.kt:51]: As before, this causes the swipe to refresh spinner is cancelled as soon as it is started.

File: app/src/main/java/com/sthoray/allright/ui/main/view/HomeFragment.kt [hunk 3]
```diff
@@ -41,6 +41,15 @@ class HomeFragment : Fragment(R.layout.fragment_home) {
...
         setOnClickListeners()
         setupObservers()
+        swipeToRefreshHomeFragment()
+    }
+
+    private fun swipeToRefreshHomeFragment(){
+        swipeRefresh.setOnRefreshListener {
+            viewModel.getRefreshFragment()
+            //Toast.makeText(context, "Page refreshed!", Toast.LENGTH_SHORT).show()
+            swipeRefresh.isRefreshing = false
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/ui/main/view/HomeFragment.kt:51]: This causes the swipe to refresh spinner is cancelled as soon as it is started. The spinner should stop only when data has successfully been fetched and is being displayed on the screen.

**Review by scsole — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/sthoray/allright/ui/main/viewmodel/MainViewModel.kt (hunk 1 above)
Comment (shaus019) [reply app/src/main/java/com/sthoray/allright/ui/main/viewmodel/MainViewModel.kt]: No worries, i have added some more comments and changed the name for this function. Hopefully it makes sense.

File: app/src/main/java/com/sthoray/allright/ui/search/view/ResultsFragment.kt (hunk 2 above)
Comment (shaus019) [reply app/src/main/java/com/sthoray/allright/ui/search/view/ResultsFragment.kt:51]: The reason line 51 here is set  to false is to make sure that spinner is cancelled after we get the updated data. If we don't do that then the spinner will spin forever even after the data has been updated.

File: app/src/main/java/com/sthoray/allright/ui/main/view/HomeFragment.kt (hunk 3 above)
Comment (shaus019) [reply app/src/main/java/com/sthoray/allright/ui/main/view/HomeFragment.kt:51]: The function on line 49 is called before the spinner is cancelled. The spinner will only cancel after we get the update data, if we don't set it to false at the end then the spinner will go forever.

Comment (scsole) [reply app/src/main/java/com/sthoray/allright/ui/main/view/HomeFragment.kt:51]: You function calls `getFeaturedCategories()`.  `getFeaturedCategories` launches a new coroutine and immediately finishes. This means your `getRefreshFragment` function does not block while network request are being made and cannot be used to signal when we are no longer refreshing. `isRefreshing` must be set back to false when we observe the response.

**Pull request closed without merging**